src/references/shared.md

```diff
@@ -0,0 +1,64 @@
+---
+minutes: 10
+existing course material:
+- basic-syntax/references.md
+- basic-syntax/references-dangling.md
+---
+
+# Shared References
+
+Rust's references provide a way to refer to another value, similar to pointers
+or C++'s references.
+
+<!-- mdbook-xgettext: skip -->
+```rust,editable
+fn main() {
+    let a = 'A';
+    let b = 'B';
+    let mut r: &char = &a;
+    println!("r: {}", *r);
+    r = &b;
+    println!("r: {}", *r);
+}
+```
+
+A reference to a type `T` has type `&T`. A reference value is made with the `&`
+operator. The `*` operator "dereferences" a reference, yielding its value.
+
+Rust will statically forbid dangling references:
+
+<!-- mdbook-xgettext: skip -->
+```rust,editable,compile_fail
+fn x_axis(x: i32) -> &(i32, i32) {
+    let point = (x, 0);
+    return &point;
+}
+```
+
+<details>
+
+* A reference is said to "borrow" the value it refers to.
+
+* References are implemented as pointers, and a key advantage is that they can
+  be much smaller than the thing they point to.
+
+* Rust does not automatically create references for you - the `&` is always
+  required.
+
+* Rust will auto-dereference in some cases, in particular when invoking
+  methods (try `r.count_ones()`). There is no need for an `->` operator
+  like in C++.
+
+* In this example, `r` is mutable so that it can be reassigned (`r = &b`).
```

Comment:
Minor suggestion (please ignore if it's not what you're looking for): might be useful to mention that this re-binds the reference (`r` becomes a pointer to something else), in contrast to C++ where an assignment to a reference changes the value of the pointee.